File: clj/src/advent_2017/day2.clj
(ns advent-2017.day2)

(defn toArr [line]
  (map #(Integer. %)
    (clojure.string/split line #"\s")))

(defn maxMinDiff [coll]
  (-
    (apply max coll)
    (apply min coll)))

(defn checksum [input]
  (->> (clojure.string/split input #"\n")
    (map toArr)
    (map maxMinDiff)
    (reduce +)))

File: js/lib/day2.js
const toArr = (line) => line.split(/\s/).map((e) => parseInt(e))

const maxMinDiff = (arr) => arr.reduce((m, e) => e > m ? e : m) - arr.reduce((m, e) => m > e ? e : m)

const checksum = (input) => input.split('\n').map(toArr).map(maxMinDiff).reduce((acc, e) => acc + e)

module.exports = {
  maxMinDiff,
  toArr,
  checksum
}
